### PR TITLE
CLN: various lgtm.com cleanups

### PR DESCRIPTION
**Pull request by simonjayhawkins**

### CODE REVIEW

**Review by WillAyd — APPROVED** [review]

File: pandas/core/arrays/integer.py [hunk 1]
```diff
@@ -78,7 +78,8 @@ def itemsize(self):
 
     @classmethod
     def construct_array_type(cls):
-        """Return the array type associated with this dtype
+        """
```
Comment (TomAugspurger) [inline pandas/core/arrays/integer.py]: Is this docstring not inherited from the base class?

**Review by TomAugspurger — APPROVED** [review]

File: pandas/core/arrays/integer.py (hunk 1 above)
Comment (simonjayhawkins) [reply pandas/core/arrays/integer.py]: not as far as I'm aware. Should it be?

Comment (simonjayhawkins) [reply pandas/core/arrays/integer.py]: It looks like `inspect.getdoc(object)` https://docs.python.org/3/library/inspect.html?highlight=inspect#inspect.getdoc could get the baseclass docstring.

I'm using VSCode and not showing baseclass docstring when no docstring exists on derived class.

`_IntegerDtype().construct_array_type.__doc__` returns None if no docstring exists on derived class.

Comment (jreback) [reply pandas/core/arrays/integer.py]: ok this is odd then I agree with @TomAugspurger that these should inherit the baseclass, odd that they are not.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks for the cleanups @simonjayhawkins

**Pull request merged**